### PR TITLE
[dagster-airflow] add `make_persistent_airflow_db_resource` 

### PR DESCRIPTION
**Pull request by Ramshackle-Jamathon**
### Summary & Motivation

This pr adds a new airflow_db resource definition called `AirflowPersistentDatabase` this can be used instead of the current default ephemeral airflow database. By persisting airflow dag run state dagster-airflow is able to support most of the remaining airflow features that don't current work with the ephemeral airflow db specifically:
- retry-from-failure in Dagster
- Using prev_execution_date in Airflow-templated DAGs
- SubDAGOperators
- Airflow Pools

### How I Tested These Changes

- unit tests

### Todo

- [x] add unit tests to cover all the newly supported airflow features

### CODE REVIEW

File: examples/with_airflow/with_airflow/__init__.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
-from .assets_repository import (
-    sda_examples as sda_examples,
-)
-from .repository import (
-    airflow_examples_repo as airflow_examples_repo,
-    task_flow_repo as task_flow_repo,
+# from .assets import (
```
Comment (schrockn) [inline examples/with_airflow/with_airflow/__init__.py:1]: rm

File: python_modules/libraries/dagster-airflow/dagster_airflow/resources/airflow_persistent_db.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            if is_airflow_2_loaded_in_environment():
+                importlib.reload(airflow.configuration)
+                importlib.reload(airflow.settings)
+                importlib.reload(airflow)
+            else:
+                importlib.reload(airflow)
+            # make sure connections are created
+            create_airflow_connections(
+                [Connection(**c) for c in context.resource_config["connections"]]
+            )
```
Comment (schrockn) [inline python_modules/libraries/dagster-airflow/dagster_airflow/resources/airflow_persistent_db.py:70]: I seem to recall very similar code in the `AirflowEphemeralDatabase`. Can these be consolidated?

File: python_modules/libraries/dagster-airflow/dagster_airflow/resources/airflow_persistent_db.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                )
+            )
+        except OverflowError:
+            raise DagsterInvariantViolationError(
+                'Date "{execution_date_str}" exceeds the largest valid C integer on the system.'
+                .format(
+                    execution_date_str=execution_date_str,
+                )
+            )
+        return execution_date
```
Comment (schrockn) [inline python_modules/libraries/dagster-airflow/dagster_airflow/resources/airflow_persistent_db.py:104]: and this is a direct copypaste job i think

**Review by schrockn — CHANGES_REQUESTED** [review]
I think there is a bunch of copy paste here we should eliminate

**Review by yuhan — APPROVED** [review]

**Comment by yuhan** [comment]
please also make sure to include `make_persistent_airflow_db_resource` in the apidocs

**Comment by Ramshackle-Jamathon** [comment]
pr for docs is here https://github.com/dagster-io/dagster/pull/12485

**Pull request merged**